src/main/kotlin/uk/gov/dluhc/printapi/mapper/StatusMapper.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.dluhc.printapi.mapper
+
+import uk.gov.dluhc.printapi.database.entity.Status.DISPATCHED
+import uk.gov.dluhc.printapi.database.entity.Status.IN_PRODUCTION
+import uk.gov.dluhc.printapi.database.entity.Status.NOT_DELIVERED
+import uk.gov.dluhc.printapi.database.entity.Status.PRINT_PROVIDER_DISPATCH_FAILED
+import uk.gov.dluhc.printapi.database.entity.Status.PRINT_PROVIDER_PRODUCTION_FAILED
+import uk.gov.dluhc.printapi.database.entity.Status.PRINT_PROVIDER_VALIDATION_FAILED
+import uk.gov.dluhc.printapi.database.entity.Status.VALIDATED_BY_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.StatusStep
+import uk.gov.dluhc.printapi.database.entity.Status as StatusEntityEnum
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.Status as StatusModelEnum
+
+class StatusMapper {
+
+    fun toStatusEntityEnum(statusStep: StatusStep, statusModelEnum: StatusModelEnum): StatusEntityEnum {
+        if (statusModelEnum == StatusModelEnum.SUCCESS) {
+            return when (statusStep) {
+                StatusStep.PROCESSED -> VALIDATED_BY_PRINT_PROVIDER
+                StatusStep.IN_MINUS_PRODUCTION -> IN_PRODUCTION
+                StatusStep.DISPATCHED -> DISPATCHED
+                else -> throw IllegalArgumentException("Undefined statusStep [$statusStep] and status [SUCCESS] combination")
```

Comment:
Could this message be more appropriate ? 
`Print status cannot be in statusStep [`NOT_MINUS_DELIVERED`] when the status is [`SUCCESS`] `